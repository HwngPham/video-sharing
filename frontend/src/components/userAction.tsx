import { useStore } from "../store";
import { Button } from "./button";

export const UserAction = () => {
  const { user } = useStore();

  return (
    <div className="flex justify-between gap-2">
      <p className="text-md text-white leading-8 mr-2">Welcome {user?.email}</p>

      <Button>Share a movie</Button>
      <Button>Logout</Button>
    </div>
  );
};
