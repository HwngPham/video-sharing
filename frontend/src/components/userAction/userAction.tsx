import { useState } from "react";
import { useStore } from "../../store";
import { Button } from "../button/button";
import { ModalSharing } from "../modalSharing/modalSharing";

export const UserAction = () => {
  const { user, logout } = useStore();
  const [isModalSharingOpen, setModalSharingOpen] = useState(false);

  const toggleModalSharing = () => setModalSharingOpen(!isModalSharingOpen);

  return (
    <div className="flex justify-between gap-2 flex-wrap">
      <p className="text-md text-white leading-8 mr-2">Welcome {user?.email}</p>

      <div className="flex gap-2">
        <Button onClick={toggleModalSharing}>Share a movie</Button>
        <Button onClick={logout}>Logout</Button>
      </div>

      <ModalSharing open={isModalSharingOpen} toggleOpen={toggleModalSharing} />
    </div>
  );
};
