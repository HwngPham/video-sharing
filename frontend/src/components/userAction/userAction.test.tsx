import { render } from "@testing-library/react";
import { UserAction } from "./userAction";

describe("<UserAction />", () => {
  it("matches snapshot", () => {
    const container = render(<UserAction />);
    expect(container).toMatchSnapshot();
  });
});
