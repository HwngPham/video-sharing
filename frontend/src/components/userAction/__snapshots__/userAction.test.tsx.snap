// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserAction /> matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-between gap-2 flex-wrap"
      >
        <p
          class="text-md text-white leading-8 mr-2"
        >
          Welcome 
        </p>
        <div
          class="flex gap-2"
        >
          <button
            class="bg-blue-500 text-white py-1 px-4 rounded-sm border cursor-pointer whitespace-nowrap hover:bg-blue-700 hover:border-transparent"
          >
            Share a movie
          </button>
          <button
            class="bg-blue-500 text-white py-1 px-4 rounded-sm border cursor-pointer whitespace-nowrap hover:bg-blue-700 hover:border-transparent"
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-between gap-2 flex-wrap"
    >
      <p
        class="text-md text-white leading-8 mr-2"
      >
        Welcome 
      </p>
      <div
        class="flex gap-2"
      >
        <button
          class="bg-blue-500 text-white py-1 px-4 rounded-sm border cursor-pointer whitespace-nowrap hover:bg-blue-700 hover:border-transparent"
        >
          Share a movie
        </button>
        <button
          class="bg-blue-500 text-white py-1 px-4 rounded-sm border cursor-pointer whitespace-nowrap hover:bg-blue-700 hover:border-transparent"
        >
          Logout
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
